src/config/config.cc

```diff
@@ -97,9 +97,9 @@ Config::Config() {
   FieldWrapper fields[] = {
       {"daemonize", true, new YesNoField(&daemonize, false)},
       {"bind", true, new StringField(&binds_, "")},
-      {"port", true, new IntField(&port, kDefaultPort, 1, PORT_LIMIT)},
+      {"port", true, new IntegerField(&port, kDefaultPort, (uint32_t)1, PORT_LIMIT)},
 #ifdef ENABLE_OPENSSL
-      {"tls-port", true, new IntField(&tls_port, 0, 0, PORT_LIMIT)},
+      {"tls-port", true, new IntegerField(&tls_port, (uint32_t)0, (uint32_t)0, PORT_LIMIT)},
```

Comment:
```suggestion
      {"tls-port", true, new IntegerField<uint32_t>(&tls_port, 0, 0, PORT_LIMIT)},
```
I prefer we specify template types explicitly instead of deduced from arguments.

Reply:
In all places (and with `int` type) or only in few places with types `uint32_t` and `uint64_t`?

Reply:
I think you can define some alias, e.g. `using IntField = IntergerField<int>`, `using UInt32Field = IntergerField<uint32_t>`.

Then you can use these alias in the config map.

Reply:
Done